src/config/chainEndpoints.ts

```diff
@@ -127,3 +128,24 @@ export const getProviderIndex = (chain: ASTAR_CHAIN) => {
       return endpointKey.ASTAR;
   }
 };
+
+// Memo: remove this function whenever individual claim  is applied in all the networks
+export const checkIsEnableIndividualClaim = () => {
```

Comment:
I think it would be better to check a node version or read some specific feature of individual claim and base on that calculate checkIsEnableIndividualClaim, rather than have logic which needs to be maintained as individual claim feature is pushed on different networks

Reply:
Sounds good. Feel free to modify this function 👍

Reply:
This method of checking for available functions in the chain doesn't seem scalable. Please refactor this

Reply:
> This method of checking for available functions in the chain doesn't seem scalable. Please refactor this

@Dinonard Could you share with me how can I query the node version that is available for the individual claim feature, please?

Reply:
Hi @Dinonard 
> V2_0_0 is the current one with unbonding period.

Is the way to check which node version is available for `individual claim` feature? Seems like both and Shiden the local node returns `v2_0_0`. 
<img width="1710" alt="image" src="https://user-images.githubusercontent.com/92044428/156402936-62085074-622a-4340-843b-fd0d64abdc17.png">

Reply:
Oh, I see.

The version will be changed to `V3_0_0` when we do runtime upgrade.
If you run local node, it will still say `V2_0_0`.

But let me change this so it's easier for you to test.
I will push an update so default version is 3.

Reply:
Thank you, @Dinonard 
btw, I found that there is no 'storageVersion' query in Astar network.

![image](https://user-images.githubusercontent.com/92044428/156493915-b9150e89-08be-409d-a71a-074208ed33f0.png)

Reply:
> This method of checking for available functions in the chain doesn't seem scalable. Please refactor this

@hoonsubin @bobo-k2 
[Fixed](https://github.com/AstarNetwork/astar-apps/pull/258/commits/a17dff706a57468b6c9bdf52654e0cf98efe96fb)

Reply:
Astar still uses legacy runtime where no `version` storage exists - you can assume for now that it's 1.

Once we upgrade Astar, it will immediately become v3.

Reply:
> Once we upgrade Astar, it will immediately become v3.

@Dinonard 
I see. Am I correct that the v3 for Astar means the network is available for the `individual claim` feature?

Reply:
`V3_0_0` means Individual claim is available, regardless of network.

In case no `Version` storage exists, this means that legacy dapps-staking is being used and the assumed version should be `V1_0_0`.

Hope this helps.